### DISCUSSION TITLE
Thoughts on encapsulation

### DISCUSSION THREAD

**Discussion by RupertAvery** [Maximilian-Winter/Bite-Programming-Language#42] [General]
Everything in Bite is public by default. Making everything private does make the language seem less "simple" by enforcing encapsulation.

I think enforcing encapsulation makes it a better "modern-ish" and more "serious" language.

**Comment by Maximilian-Winter** [comment]
I'm already parsing private and public modifiers and make everything private by default. But right now this isnt implemented.

**Comment by Maximilian-Winter** [comment]
So you can access everything right now

**Comment by Maximilian-Winter** [comment]
I'm planning on adding this soon

**Comment by vorov2** [comment]
There is always a Python way to do it, e.g. to use special notation for privates like `__private__`. Objectively speaking this is still as public as everything else and not encapsulated, but you also state by this notation that it is not included in a contract, therefore code that directly references these "privates" can break in future. That is an approach that allows to "keep it simple".
